### PR TITLE
[soundcloud] Fix paged playlist download archival

### PR DESCRIPTION
**Pull request by CoryDHall**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

rel: #19022 

Paged playlists do not work as intended with `--download-archive`. This attaches missing data by considering additional information already present in the collection.

### CODE REVIEW

File: youtube_dl/extractor/soundcloud.py [hunk 1]
```diff
@@ -397,13 +397,26 @@ def resolve_permalink_url(candidates):
...
+                    (e, tr_dict, pl_dict))
                 if permalink_url:
-                    entries.append(self.url_result(permalink_url, video_id=entry_id))
+                    entry_data = self.url_result(
+                        permalink_url,
+                        video_id=entry_id, video_title=entry_title)
+
+                    if isinstance(tr_dict, dict):
+                        # if entry has track data, attach extractor key
+                        entry_data['ie_key'] = SoundcloudIE.ie_key()
```
Comment (dstftw) [inline youtube_dl/extractor/soundcloud.py]: This must be passed in `url_result`.

File: youtube_dl/extractor/soundcloud.py [hunk 2]
```diff
@@ -397,13 +397,26 @@ def resolve_permalink_url(candidates):
...
                         permalink_url = cand.get('permalink_url')
                         entry_id = self._extract_id(cand)
+                        title = cand.get('title')
                         if permalink_url and permalink_url.startswith('http'):
-                            return permalink_url, entry_id
+                            return permalink_url, entry_id, title
 
             for e in collection:
-                permalink_url, entry_id = resolve_permalink_url((e, e.get('track'), e.get('playlist')))
+                tr_dict = e.get('track')
```
Comment (dstftw) [inline youtube_dl/extractor/soundcloud.py]: Inline.

File: youtube_dl/extractor/soundcloud.py [hunk 3]
```diff
@@ -397,13 +397,26 @@ def resolve_permalink_url(candidates):
...
+                pl_dict = e.get('playlist')
```
Comment (dstftw) [inline youtube_dl/extractor/soundcloud.py]: Inline.

**Comment by CoryDHall** [comment]
@dstftw 

Sorry for the sloppy code, I'm not super familiar with python conventions, and some of the patterns in this extractor are kind of strange

File: youtube_dl/extractor/soundcloud.py [hunk 4]
```diff
@@ -402,19 +402,17 @@ def resolve_permalink_url(candidates):
                             return permalink_url, entry_id, title
 
             for e in collection:
-                tr_dict = e.get('track')
-                pl_dict = e.get('playlist')
+                extractor = None
+                if isinstance(e.get('track'), dict):
```
Comment (dstftw) [inline youtube_dl/extractor/soundcloud.py]: This does not guarantee you anything. What you should do is to check `suitable` on resolved URL.

File: youtube_dl/extractor/soundcloud.py [hunk 5]
```diff
@@ -397,13 +397,21 @@ def resolve_permalink_url(candidates):
...
+                            return permalink_url, entry_id, title
 
             for e in collection:
-                permalink_url, entry_id = resolve_permalink_url((e, e.get('track'), e.get('playlist')))
+                permalink_url, entry_id, entry_title = resolve_permalink_url(
+                    (e, e.get('track'), e.get('playlist')))
+
                 if permalink_url:
-                    entries.append(self.url_result(permalink_url, video_id=entry_id))
+                    entry_data = self.url_result(
```
Comment (dstftw) [inline youtube_dl/extractor/soundcloud.py]: Inline.

File: youtube_dl/extractor/soundcloud.py [hunk 6]
```diff
@@ -397,13 +397,21 @@ def resolve_permalink_url(candidates):
                     if isinstance(cand, dict):
                         permalink_url = cand.get('permalink_url')
                         entry_id = self._extract_id(cand)
+                        title = cand.get('title')
                         if permalink_url and permalink_url.startswith('http'):
-                            return permalink_url, entry_id
+                            return permalink_url, entry_id, title
```
[the remark this answers is not in the document]
Comment (CoryDHall) [reply youtube_dl/extractor/soundcloud.py]: @dstftw I thought so too, but again, I'm not strong in Python, and I'm new to the codebase, so I did not want to do any refactoring if it means this bug can be fixed without it

**Pull request closed without merging**